### DISCUSSION TITLE
Ios in offline mode is not working

### DISCUSSION THREAD

**Discussion by mlyeung001** [serwist/serwist#71] [Q&A]
I follow the guide to deploy a PWA in my app. I tried the PWA in android, ios and chrome in desktop. Android and chrome are working in offline, but ios is not wokring in offline. Why?

I am using ios 17 for the test.

here is my code
In next.config.js
`const withSerwist = require('@serwist/next').default({
  swSrc: 'src/pwa/sw.ts',
  swDest: 'public/sw.js',
  reloadOnOnline: true
})
module.exports = withSerwist({
...
});
`

In src/pwa/sw.ts,
```
import { defaultCache } from '@serwist/next/browser'
import type { PrecacheEntry } from '@serwist/precaching'
import { installSerwist } from '@serwist/sw'

declare const self: ServiceWorkerGlobalScope & {
  __SW_MANIFEST: (PrecacheEntry | string)[] | undefined
}

installSerwist({
  precacheEntries: self.__SW_MANIFEST,
  skipWaiting: true,
  clientsClaim: true,
  navigationPreload: true,
  runtimeCaching: defaultCache
})
```

**Comment by DuCanhGH** [comment]
@mlyeung001 this is too vague for me to know what's exactly your problem. Having read between the lines and made some assumptions, I'm guessing that this is simply because your page wasn't cached. The default behaviour for `@serwist/next` is that no pages are precached and that pages are only cached as they are fetched (runtime caching). This means that you have to visit these pages first before they are available to the offline mode. If you'd like to change that, you need to add some pages to the precache manifest through [`additionalPrecacheEntries`](https://serwist.pages.dev/docs/build/configuring/base-options/additional-precache-entries) (it is recommended that you use some build identification for a revision).

I don't exactly know what your issue is, so I can't say for sure. Then again, Apple is notoriously shitty to PWAs, so sometimes things just don't work out.
